src/server.rs

```diff
@@ -127,7 +127,15 @@ macro_rules! messages {
                 });
             }
 
-            let id = ls_command.get("id").and_then(|id| id.as_u64()).map(|id| id as usize);
+            let id_value = ls_command.get("id").unwrap();
+            let mut id: Option<usize> = None;
+            if id_value.is_u64() {
+                id = Some(ls_command.get("id").unwrap().as_u64().unwrap() as usize);
+            } else if id_value.is_string() {
+                id = Some(usize::from_str_radix(id_value.as_str().unwrap(), 10).unwrap());
+            } else {
+                return Err(ParseError::new(ErrorKind::InvalidData, "Id is not a number or numeric string", id))
```

Comment:
I think we should not return early here, since if we don't need the id, we might be able to process the message and ignore the id. Where we currently return an error, you could add the 'numeric string' bit.